src/mrinufft/operators/autodiff.py

```diff
@@ -62,7 +63,7 @@ def backward(ctx, dy):
                     dim=0,
                 ),
                 dim=0,
-            ).type_as(traj)
+            ).to(ctx.dtype)
         return grad_data, grad_traj, None
```

Comment:
you can use  `nufft_op.dtype` or `nufft_op.cpx_dtype`

Reply:
Good catch

Reply:
Nope, reverting this :P  https://github.com/mind-inria/mri-nufft/actions/runs/10054335386/job/27788758132?pr=156
Turns out we need a torch.dtype and torch does not understand numpy dtype:

[1] https://github.com/pytorch/pytorch/issues/40568
[2] https://discuss.pytorch.org/t/converting-a-numpy-dtype-to-torch-dtype/52279

Reply:
But we have NP2TORCH exactly for this (in `mrinufft._utils`) ;)